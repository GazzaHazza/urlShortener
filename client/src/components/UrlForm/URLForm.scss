@import './../../styles/_colors.scss';

.url-form {
    &__container {
        padding: 20px;
        background: $color1;
    }
}