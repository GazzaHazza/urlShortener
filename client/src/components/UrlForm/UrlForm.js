import React, { Component } from "react";
import PropTypes from "prop-types";
import "./UrlForm.css";
class UrlForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      urlValue: ""
    };
  }
  handleInputChange = e => {
    this.setState({ urlValue: e.target.value });
  };
  handleOnClick = () => {
    this.props.onClickGenerate(this.state.urlValue);
  };
  render() {
    const { isBusy } = this.props;
    const buttonText = isBusy ? "Generating..." : "Generate short link";
    return (
      <div className="url-form">
        <div className="url-form__container">
          <input
            type="text"
            className="url-form__input"
            value={this.state.urlValue}
            onChange={this.handleInputChange}
          />
          <button className="url-form__button" onClick={this.handleOnClick}>
            {buttonText}
          </button>
        </div>
      </div>
    );
  }
}
UrlForm.propTypes = {
  isBusy: PropTypes.bool,
  onClickGenerate: PropTypes.func
};
UrlForm.defaultProps = {
  isBusy: false
};
export default UrlForm;
