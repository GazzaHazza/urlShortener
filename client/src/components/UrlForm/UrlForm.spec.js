import React from "react";
import { shallow, mount } from "enzyme";
import UrlForm from "./UrlForm";

it("renders correctly without isBusy prop", () => {
  const wrapper = shallow(<UrlForm />);
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly with isBusy prop", () => {
  const wrapper = shallow(<UrlForm isBusy={true} />);
  expect(wrapper).toMatchSnapshot();
});

it("set correct default prop", () => {
  const wrapper = mount(<UrlForm />);
  expect(wrapper.prop("isBusy")).toBe(false);
});

it("calls onClickGenerate with correct data", () => {
  const onClickGenerateStub = jest.fn();
  const wrapper = shallow(<UrlForm onClickGenerate={onClickGenerateStub} />);
  wrapper
    .find("input")
    .simulate("change", { target: { value: "http://www.test.com" } });
  wrapper.find("button").simulate("click");
  expect(onClickGenerateStub).toBeCalledWith("http://www.test.com");
});

it("handleInputChange shuold set state correct", () => {
  const wrapper = shallow(<UrlForm />);
  wrapper
    .find("input")
    .simulate("change", { target: { value: "http://www.test.com" } });
  expect(wrapper.state("urlValue")).toBe("http://www.test.com");
});
