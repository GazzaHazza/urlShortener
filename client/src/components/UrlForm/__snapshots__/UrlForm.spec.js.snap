// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with isBusy prop 1`] = `
<div
  className="url-form"
>
  <div
    className="url-form__container"
  >
    <input
      className="url-form__input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="url-form__button"
      onClick={[Function]}
    >
      Generating...
    </button>
  </div>
</div>
`;

exports[`renders correctly without isBusy prop 1`] = `
<div
  className="url-form"
>
  <div
    className="url-form__container"
  >
    <input
      className="url-form__input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="url-form__button"
      onClick={[Function]}
    >
      Generate short link
    </button>
  </div>
</div>
`;
